aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotWebSocketUrlSigner.java

```diff
@@ -57,18 +57,43 @@ public AWSIotWebSocketUrlSigner(String serviceName) {
     /**
      * Given the region and service name provided to the client, the endpoint and the current time
      * return a signed connection URL to be used when connecting via WebSocket to AWS IoT.
+     *
+     * @param endpoint service endpoint with or without customer specific URL prefix.
+     * @param awsCredentials credential set to be used in SigV4 signature algorithm.
+     * @param currentTimeInMillis time value to be used in SigV4 calculations.  In milliseconds.
+     * @return a URL with SigV4 signature formatted to be used with AWS IoT.
+     */
+    String getSignedUrl(final String endpoint,
+                        final AWSCredentials awsCredentials,
+                        final long currentTimeInMillis) {
+        return getSignedUrl(endpoint, awsCredentials, currentTimeInMillis, null);
```

Comment:
Instead of passing `null`, and then checking if its null, just ensure that it's  never null, to begin.

To achieve this, you can have this line look like:
```
Region endpointRegion = AwsIotEndpointUtility.getRegionFromIotEndpoint(endpoint);
return getSignedUrl(endpoint, awsCredentials, currentTimeInMillis, endpointRegion);
```

Then, in your new overloaded version of this method, just always use the `region`, without any conditional logic.